capstone/src/main/java/com/google/sps/servlets/student/StudentServlet.java

```diff
@@ -91,16 +88,31 @@ private static String convertToJsonUsingGson(StudentInfo info) {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: All commented steps below
-    // 1. Add club to logged in student's club list if requested
-    // 2. Remove club from logged in student's club list if requested
-    // 3. Update student information with edited content
+    // 1. Remove club from logged in student's club list if requested
+    // 2. Update student information with edited content
 
     // No need to check if user's information is not in Datastore - this is done in doGet method
     // Get student object based on the logged in email
     UserService userService = UserServiceFactory.getUserService();
     String userEmail = userService.getCurrentUser().getEmail();
 
-    response.sendRedirect("/profile.html");
+    Query query = new Query(userEmail);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Entity student = ImmutableList.copyOf(results.asIterable()).get(0);
```

Comment:
Should we check there are valid student entries before calling .get(0)?

Reply:
I believe the student should always be valid in the doPost method since the doGet method creates entities if necessary but I agree that it's best to check anyway so just added a helper method to check for this!